### ISSUE TITLE
kubectl logs/run fails with Authorization error with custom VPC

### ISSUE THREAD

**Issue by SamLR** [weaveworks/eksctl#4645]
**What were you trying to accomplish?**

Trying to get logs from a pod fails with Authorization Error

**What happened?**

When running `kubectl run` and `kubectl logs` on an EKS cluster with custom VPC the command fails with an error like.

```bash
$ kubectl run test -ti --rm --image=alpine
If you don't see a command prompt, try pressing enter.
Error attaching, falling back to logs: unable to upgrade connection: Authorization error (user=kube-apiserver-kubelet-client, verb=create, resource=nodes, subresource=proxy)
pod "test" deleted
Error from server (InternalError): Internal error occurred: Authorization error (user=kube-apiserver-kubelet-client, verb=get, resource=nodes, subresource=proxy)
```

The error response seems to take a long time (~60s) so appears to be a timeout but is returned as an authorization error (despite `kubectl auth can-i get pod/logs` returning `yes`).

The cluster is configured with `managedNodeGroup` with `privateNetworking`. The VPC is custom and configured with `enableDnsSupport: true` and `enableDnsHostnames: false` with the nodes being deployed in the private subnets. The vpc has public and private endpoints and uses `publicAccessCIDRs` set to my IP. 

**Versions**
```bash
$ eksctl info
eksctl version: 0.77.0
kubectl version: v1.21.5
OS: darwin
```

**Comment by SamLR** [comment]
**tl;dr** fix: setting `enableDnsHostnames: true` on the VPC fixed it for me.

Also added this to the troubleshooting docs in the PR

**Comment by aclevername** [comment]
thanks for the summary and great work @SamLR !
